apps/login/src/lib/service.ts

```diff
@@ -108,3 +109,20 @@ export function getServiceUrlFromHeaders(headers: ReadonlyHeaders): {
     serviceUrl: instanceUrl,
   };
 }
+
+export function constructUrl(request: NextRequest, path: string) {
+  const forwardedProto = request.headers.get("x-forwarded-proto") ?? "https";
+
+  const forwardedHost =
+    request.headers.get("x-zitadel-forward-host") ??
+    request.headers.get("host");
```

Comment:
Let's support the common x-forwarded-host 

```suggestion
  const forwardedHost =
    request.headers.get("x-zitadel-forward-host") ??
    request.headers.get("x-forwarded-host") ??
    request.headers.get("host");
```